src/js/client/editprefset-settingsTable-widget.js

```diff
@@ -0,0 +1,324 @@
+            func: "{that}.filterSettings"
+        },
+        allSettingsEnabled: {
+            func: "{that}.filterSettings"
+        },
+        flatPrefs: {
+            func: "{that}.reRender",
+            excludeSource: ["init"]
+        }
+    },
+    bindings: { // Binding selectors to model paths
+    },
+    selectors: {
+        valueDisplayCell: ".pmt-value-display",
+        enabledBooleanInputs: ".pmt-enabled-boolean",
+        settingsFilter: "#pmt-filter-container",
+        settingsRows: ".pmt-settings-table-row",
+        addContextButton: ".pmt-add-context-button",
+        mineAllSwitchContainer: "#pmt-mineAllSwitch-container",
+        removeProductButton: ".pmt-remove-product",
+
+        // PSP items
+        pspShowCheckboxes: ".pmt-psp-show-checkbox",
+        pspMemoryCheckboxes: ".pmt-psp-memory-checkbox"
+    },
+    templates: {
+        initial: "editprefset-settingsTable-widget"
+    },
+    components: {
+        mineAllSwitch: {
+            type: "gpii.devpmt.mineAllSwitch.default",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.mineAllSwitchContainer",
+            options: {
+                selectors: {
+                    initial: "#pmt-mineAllSwitch-area"
+                },
+                model: {
+                    settingsFilter: "{settingsTableWidget}.model.allSettingsEnabled",
+                    allSettingsCount: "{settingsTableWidget}.model.termUsage.all",
+                    mySettingsCount: "{settingsTableWidget}.model.termUsage.npset"
+                }
+            }
+        },
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.settingsFilter",
+            options: {
+                selectors: {
+                    initial: "#pmt-common-filter-area"
+                },
+                model: {
+                    filterText: "{settingsTableWidget}.model.settingsFilter"
+                }
+            }
+        }
+    },
+    invokers: {
+        filterInit: {
+            funcName: "gpii.devpmt.settingsTable.filterInit",
+            args: ["{that}"]
+        },
+        // Overriding renderInitialMarkup to take `that` as a render context.
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}"]
+        },
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.settingsTable.updateLunrIndex",
+            args: ["{that}"]
+        },
+        filterSettings: {
+            funcName: "gpii.devpmt.settingsTable.filterSettings",
+            args: ["{that}", "{that}.dom.settingsRows", "{that}.lunrIndex", "{that}.model.settingsFilter"]
+        },
+        enableProductListener: {
+            funcName: "gpii.devpmt.settingsTable.enableProductListener",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"] // event
```

Comment:
Please sync up the function argument name used for referencing "{gpii.devpmt.editPrefs}". For example, the `enableProductListener()` function uses "devpmt" while `removeProduct()` uses "editPrefs". This comment applies to the entire pull request.

Reply:
fixed up